CRM/Contract/Form/Create.php

```diff
@@ -13,9 +13,15 @@ function buildQuickForm(){
 
 
     $this->cid = CRM_Utils_Request::retrieve('cid', 'Integer');
+    if (empty($this->cid)) {
+      $this->cid = $this->get('cid');
```

Comment:
@michaelmcandrew When you've submitted the create form, cid is only in $this->get and not request.  I added a check to make sure cid is specified so the form doesn't give a fatal error.